app/models/manageiq/providers/hawkular/middleware_manager/event_catcher/runner.rb

```diff
@@ -63,14 +66,28 @@ def event_monitor_handle
     @event_monitor_handle ||= ManageIQ::Providers::Hawkular::MiddlewareManager::EventCatcher::Stream.new(@ems)
   end
 
-  def valid?(event)
-    pattern = INCLUDE_EVENTS[event.category]
+  def whitelist?(event)
+    return false unless @whitelist.key?(event.category)
+    pattern = @whitelist[event.category]
+    return true unless pattern
     message = event.context['Message']
-    pattern && message && message.match(pattern)
+    message && message.match(pattern)
   end
 
-  # Check for blacklisted event type
-  def filtered?(event_data)
-    filtered_events.include?(event_data[:event_type])
+  def blacklist?(event_type)
+    filtered_events.include?(event_type)
+  end
+
+  def event_to_hash(event, event_type, ems_id = nil)
+    {
+      :ems_id          => ems_id,
+      :source          => 'HAWKULAR',
+      :timestamp       => Time.zone.at(event.ctime / 1000),
+      :event_type      => event_type,
+      :message         => event.context['Message'],
+      :middleware_ref  => event.context['CanonicalPath'],
```

Comment:
middleware_ref is not intended for the db.  It is the middleware ems ref (the key) used to determine the middleware_id, which is then stored in the db. See ems_event.process_middleware_entities_in_event for the details.